pkg/hostpath/nodeserver.go

```diff
@@ -90,18 +90,17 @@ func (ns *nodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublis
 		options = append(options, "ro")
 	}
 	mounter := mount.New("")
-	path := provisionRoot + volumeId
-	newEphemeral := false
+
+	path := getVolumePath(volumeId)
 	if ns.ephemeral {
-		if err = os.MkdirAll(path, 0777); err != nil && !os.IsNotExist(err) {
+		volPath, err := createVolumeDir(volumeId)
+		if err != nil && !os.IsExist(err) {
 			return nil, status.Error(codes.Internal, err.Error())
 		}
-		newEphemeral = true
+		glog.V(4).Infof("ephemeral mode: created volume: %s", volPath)
 	}
+
 	if err := mounter.Mount(path, targetPath, "", options); err != nil {
-		if newEphemeral {
-			os.RemoveAll(path)
-		}
```

Comment:
You removed the code here that cleans up volumes if they fail to allocate properly. I think this can leave behind volume directories that will never be touched again on error? I don't think the NodeUnpublishVolume is ever called if NodePublishVolume fails.

Reply:
@kfox1111 I did not quite follow the rationale for the extra check.  I see what you mean now.  I will add that check back in there.